nextcord/guild.py

```diff
@@ -2159,6 +2161,82 @@ async def fetch_channel(self, channel_id: int, /) -> Union[GuildChannel, Thread]
         channel: GuildChannel = factory(guild=self, state=self._state, data=data)  # type: ignore
         return channel
 
+    async def onboarding(self) -> Onboarding:
+        """|coro|
+
+        Retrieves the onboarding for this guild.
+
+        Raises
+        ------
+        Forbidden
+            You do not have proper permissions to get the information.
+        HTTPException
+            An error occurred while fetching the information.
+
+        Returns
+        -------
+        :class:`Onboarding`
+            The onboarding object for this guild.
+        """
+        data = await self._state.http.get_onboarding(self.id)
+        return Onboarding(guild=self, data=data)
+
+    async def edit_onboarding(
+        self,
+        *,
+        reason: Optional[str] = None,
+        prompts: List[OnboardingPrompt] = MISSING,
+        default_channel_ids: List[Snowflake] = MISSING,
+        enabled: bool = MISSING,
+        mode: OnboardingMode = MISSING,
+    ) -> Onboarding:
+        """|coro|
+
+        Edits the onboarding of the guild.
+
+        You must have the :attr:`~Permissions.manage_guild` &
+        :attr:`~Permissions.manage_roles` permissions to use this.
+
+        .. versionadded:: 3.0
+
+        Parameters
+        ----------
+        reason: Optional[:class:`str`]
+            The reason for doing this action. Shows up on the audit log.
+        prompts: List[:class:`OnboardingPrompt`]
+            The screen's prompts.
+        default_channel_ids: List[:class:`int`]
+            The list of the IDs of channels that users are opted into immediately.
+        enabled: :class:`bool`
+            Whether this screen is enabled.
+        mode: :class:`OnboardingMode`
+            The criteria needed for onboarding to be enabled.
+
+        Raises
+        ------
+        Forbidden
+            You do not have permission to edit the onboarding.
+        HTTPException
+            Editing the onboarding failed.
+
+        Returns
+        -------
+        :class:`Onboarding`
+            The onboarding object for this guild.
+        """
+        payload = {}
+        if prompts is not MISSING:
```

Comment:
A prompt or enum is not json serialisable (at least to what you would need - enums need `.value`). Could you please properly implement this and test it yourself.